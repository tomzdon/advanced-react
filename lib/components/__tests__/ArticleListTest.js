import React from "react";
import ArticleList from "../ArticleList";

import renderer from "react-test-renderer";

const testProps = {
  articles: {
    a: { id: "a" },
    b: { id: "b" },
  },
  articleActions: {
    lookupAuthor: jest.fn(() => ({})),
  },
};

describe("ArticleList", () => {
  it("renderers correctly", () => {
    const tree = renderer.create(<ArticleList {...testProps} />).toJSON();

    expect(tree).toMatchSnapshot();
    expect(tree.children.length).toBe(2);
  });
});
