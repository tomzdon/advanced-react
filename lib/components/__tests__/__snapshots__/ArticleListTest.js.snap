// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renderers correctly 1`] = `
<div>
  <div>
    <div />
    <div />
    <div>
      <a
        href={undefined}
      >
         
      </a>
    </div>
    <div />
  </div>
  <div>
    <div />
    <div />
    <div>
      <a
        href={undefined}
      >
         
      </a>
    </div>
    <div />
  </div>
</div>
`;
